WooCommerce/Classes/ViewRelated/BottomSheet/ListSelector/BottomSheetListSelectorViewController.swift

```diff
@@ -0,0 +1,127 @@
+    }
+
+    // MARK: UITableViewDataSource
+    //
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return dataSource.data.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: rowType.reuseIdentifier,
+                                                       for: indexPath) as? Cell else {
+                                                        fatalError()
+        }
+        let model = dataSource.data[indexPath.row]
+        dataSource.configureCell(cell: cell, model: model)
+
+        return cell
+    }
+
+    // MARK: UITableViewDelegate
+    //
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+
+        let selected = dataSource.data[indexPath.row]
+        if selected != dataSource.selected {
+            dataSource.handleSelectedChange(selected: selected)
+            tableView.reloadData()
+        }
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        guard let header = tableView.dequeueReusableHeaderFooterView(withIdentifier: BottomSheetListSelectorSectionHeaderView.reuseIdentifier)
+            as? BottomSheetListSelectorSectionHeaderView else {
+                fatalError()
+        }
+        header.configure(text: viewProperties.title)
+        return header
+    }
+
+    func tableView(_ tableView: UITableView, estimatedHeightForHeaderInSection section: Int) -> CGFloat {
+        return estimatedSectionHeight
+    }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return UITableView.automaticDimension
+    }
+}
+
+// MARK: - View Configuration
+//
+private extension BottomSheetListSelectorViewController {
+
+    func configureMainView() {
+        view.backgroundColor = .listForeground
+    }
+
+    func configureTableView() {
+        tableView.delegate = self
+        tableView.dataSource = self
+
+        tableView.rowHeight = UITableView.automaticDimension
+        tableView.backgroundColor = .listForeground
+
+        registerTableViewCells()
+        registerTableViewHeaderFooters()
+
+        tableView.removeLastCellSeparator()
+        tableView.separatorStyle = .none
+        tableView.bounces = false
+    }
+
+    func registerTableViewCells() {
+        tableView.register(rowType.loadNib(), forCellReuseIdentifier: rowType.reuseIdentifier)
+    }
+
+    func registerTableViewHeaderFooters() {
+        let headersAndFooters = [BottomSheetListSelectorSectionHeaderView.self]
+
+        for kind in headersAndFooters {
+            tableView.register(kind.loadNib(), forHeaderFooterViewReuseIdentifier: kind.reuseIdentifier)
+        }
```

Comment:
It looks like we can simplify this for now. Well, until we have more headers and footers, which is probably unlikely. 😄 

```suggestion
        let type = BottomSheetListSelectorSectionHeaderView.self
        tableView.register(type.loadNib(), forHeaderFooterViewReuseIdentifier: type.reuseIdentifier)
```

Reply:
yup! updated in: https://github.com/woocommerce/woocommerce-ios/pull/2155/commits/f657b4eed645b650bcfc9621b14f04ca07056681